src/ui/inventory.js

```diff
@@ -61,12 +63,32 @@ export class Inventory {
     addToInventory(item, createSprite = true) {
         // Search for empty slot
         for (let index = 0; index < this.playerItems.length; index++) {
-            // Check if object exists
-            if (!Object.keys(this.playerItems[index]).length) {
+            let itemExists = Object.keys(this.playerItems[index]).length;
+
+            // Check if it can stack with other items
+            if (
+                itemExists &&
+                item.stackable &&
+                this.playerItems[index].item == item.item &&
+                this.playerItems[index].type == item.type
+            ) {
+                let curItem = this.inventory[index];
+
+                // Update the item in the game
+                curItem.setNumItems(curItem.numItems + item.numItems);
+
+                // Update it in the cookies
+                this.playerItems[index].count += item.numItems;
+
+                return true;
+            }
+            // Check if slot is empty
+            else if (!itemExists) {
                 this.addToInventoryAtIndex(item, index, createSprite);
                 return true;
             }
```

Comment:
Not sure how JS works but should we explicitly return false here?

Reply:
I'm returning false specifically when the item cannot be added because the inventory is full. Which I do at line 76/98. Not sure what you're talking about here

Reply:
I was hoodwinked by the hidden code - still not used to github prs